pkg/core/types.go

```diff
@@ -82,3 +82,23 @@ func (d *deployment) SetPodTemplate(template *corev1.PodTemplateSpec) {
 func (d *deployment) DeepCopy() podController {
 	return &deployment{d.Deployment.DeepCopy()}
 }
+
```

Comment:
It would be nice to have some comments through the `podController` stuff that's been added recently but we don't have any right now so can leave them til later